### PR TITLE
Terminal adapter

### PR DESCRIPTION
**Pull request by keathley**
Follow up to #8 and #10. 
### TL;DR

I've further refactored the slack adapter and added a console adapter to make it easier for people to do local development.
# Why

We need a better way to do development with Butler. For instance if someone is working on a plugin then they shouldn't have to get slack credentials and such just to work on it.
# How

I've refactored everything specific to the slack adapter out of the `Butler.Bot` module and into a specific Slack module. This means that the Bot implementation is much simpler. The Bot now serves as a dispatcher between the plugins and adapter.

The Console adapter just uses `IO.gets/1` and `IO.puts/0` to get data from the user and to output data to the user.

I've updated the application config to use the console adapter by default. We'll need to update the documentation to explain how to choose between different adapters and how to set them up. It would also be good to eventually add specific development and production configs which could use the console adapter for development and the slack adapter for production.

Let me know what you think.

### CODE REVIEW

File: lib/butler/adapters/console.ex [hunk 1]
```diff
@@ -0,0 +1,51 @@
+defmodule Butler.Adapters.Console do
+  require Logger
+
+  def start_link(_opts \\ []) do
```
Comment (bbrock25) [inline lib/butler/adapters/console.ex:4]: what is this `_opts` used for?

**Comment by bbrock25** [comment]
very cool, how do I run this locally to check it out?

File: lib/butler/adapters/console.ex (hunk 1 above)
Comment (bbrock25) [inline lib/butler/adapters/console.ex:4]: nvm. found it in the genserver docs..

**Comment by keathley** [comment]
@bbrock25 You can pull down the branch and run `mix run --no-halt` to try it out locally.

File: lib/butler/adapters/console.ex (hunk 1 above)
Comment (keathley) [inline lib/butler/adapters/console.ex:4]: Yup. Just generic params that some of the adapters might need to have. We don't need them for the console adapter hence the `_`

**Comment by bbrock25** [comment]
https://www.dropbox.com/s/s3didw6yotsfjm9/Screenshot%202015-09-20%2018.42.04.png?dl=0

LGTM :+1:
